spec/support/site_settings_helpers.rb

```diff
@@ -8,6 +8,10 @@ def new_settings(provider)
       # so we set listen_for_changes to false
       self.listen_for_changes = false
       self.provider = provider
+
+      def self.setting(*args)
+        super
+      end
```

Comment:
This is required and OK because this is a new class meant for testing and does not alter `SiteSetting` which is considered a global that holds states.

Reply:
Hmm.. just to confirm on functionality, this helper currently allows tests to have their own provider, right?

Reply:
This method creates a new class that "acts like SiteSetting" by importing the `SiteSettingExtension` module. It is mainly used for us to unit test methods in `SiteSettingExtension`.